backend/src/GraphQl/AccessRights/DataAccessRightsMutations.cs

```diff
@@ -0,0 +1,212 @@
+        //    )
+        //)
+        //    return new AddDataAccessRightsPayload(
+        //        new AddDataAccessRightsError(
+        //            AddDataAccessRightsErrorCode.UNAUTHORIZED,
+        //            $"The current user is not authorized to approval for the institution.",
+        //            []
+        //        )
+        //    );
+
+        var accessRights = await context.AccessRights.FirstOrDefaultAsync(x => x.InstitutionId == input.InstitutionId, cancellationToken).ConfigureAwait(false);
+
+        if (accessRights is not null)
+        {
+            return new AddAccessRightsPayload(
+                new AddAccessRightsError(
+                    AddAccessRightsErrorCode.ALREADY_EXISTS,
+                    $"The access rights for this institution already exist.",
+                    []
+                )
+            );
+        }
+        else
+        {
+            accessRights = new Data.AccessRights(
+                input.InstitutionId,
+                input.AllowedUserCount,
+                input.AllowedDatasetsPerTimeSpan,
+                input.Period);
+            context.AccessRights.Add(accessRights);
+        }
+
+        await context.SaveChangesAsync(cancellationToken).ConfigureAwait(false);
+        return new AddAccessRightsPayload(accessRights);
+    }
+
+    public async Task<AddAccessRightsPayload> UpdateAccessRightsAsync(
+        AccessRightsInput input,
+        ApplicationDbContext context,
+        AppSettings appSettings,
+        IHttpClientFactory httpClientFactory,
+        IHttpContextAccessor httpContextAccessor,
+        IUserService userService,
+        IDataService dataService,
+        CancellationToken cancellationToken
+    )
+    {
+        var currentUser = await userService.GetCurrentUser(
+            cancellationToken).ConfigureAwait(false);
+        if (currentUser == null)
+        {
+            return new AddAccessRightsPayload(
+                new AddAccessRightsError(
+                    AddAccessRightsErrorCode.UNAUTHENTICATED,
+                    $"The user is not authenticated.",
+                    []
+                )
+            );
+        }
+
+        //if (!CommonAuthorization.(
+        //    currentUser,
+        //    input.CreatorId,
+        //    appSettings,
+        //    httpClientFactory,
+        //    httpContextAccessor,
+        //    cancellationToken
+        //    )
+        //)
+        //    return new AddDataAccessRightsPayload(
+        //        new AddDataAccessRightsError(
+        //            AddDataAccessRightsErrorCode.UNAUTHORIZED,
+        //            $"The current user is not authorized to approval for the institution.",
+        //            []
+        //        )
+        //    );
+
+        var accessRights = await context.AccessRights.FirstOrDefaultAsync(x => x.InstitutionId == input.InstitutionId, cancellationToken).ConfigureAwait(false);
+
+        if (accessRights is not null)
```

Comment:
Could you remodel this check such that the if-branch is the current else-branch (which returns) and the current body of the if-branch just comes before the `SaveChangesAsync` below and is not wrapped in a `else` (because that is unnecessary then).